### PR TITLE
Enhance/#7574 - Implement `createCustomDimension` fetch store

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #7574 

## Relevant technical choices

* The IB states to add a `createCustomDimension` action. However, we no longer need it. A new `createCustomDimensions` (with an “s”) action will be defined in [7574](https://github.com/google/site-kit-wp/issues/7598).

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/datastore/custom-dimensions.js
Comment (aaemnnosttv) [inline assets/js/modules/analytics-4/datastore/custom-dimensions.js:50]: This also isn't required by the AC, but also, while all of the custom dimensions we need now will be event scoped, we shouldn't limit the action to only creating such dimensions. Later if we did have such a requirement, it would be better to fail the request rather than silently create it as event-scoped instead.

File: assets/js/modules/analytics-4/datastore/custom-dimensions.js
Comment (aaemnnosttv) [inline assets/js/modules/analytics-4/datastore/custom-dimensions.js:53]: This seems reasonable but I'm not sure about this state since we're going to keep track of available custom dimensions in a module setting. I can see how this could be useful, I'm just not aware of a plan for its use. @nfmohit ?

We could potentially add the parameter name to the list of available dimensions in the setting to unblock things that will depend on that state sooner, otherwise those would be blocked until we sync the available dimensions (which is where we'd refresh the persisted list).

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @hussain-t – a few points to address here, but also, let's add a test for this slice as mentioned in the IB. I think is still worth doing even though we aren't creating the `createCustomDimension` action. We'll definitely have other tests to add in future issues, so I think it makes sense to scaffold here, but even a few tests for the fetch action would be great. We don't normally test these but there are a few instances where we do.

File: assets/js/modules/analytics-4/datastore/custom-dimensions.js
Comment (hussain-t) [reply assets/js/modules/analytics-4/datastore/custom-dimensions.js:50]: Thanks for the feedback. The decision to hardcode the `scope` as `EVENT` was made following a discussion with @nfmohit during the IB phase. However, in light of your feedback and considering future flexibility:

1. The backend logic has been implemented to handle cases where the `scope` might not be set. In such instances, it defaults to `EVENT`.
2. If a `scope` value is provided, it ensures that it's validated against the [DimensionScope](https://developers.google.com/analytics/devguides/config/admin/v1/rest/v1beta/properties.customDimensions#DimensionScope) enum values.

File: assets/js/modules/analytics-4/datastore/custom-dimensions.js
Comment (nfmohit) [reply assets/js/modules/analytics-4/datastore/custom-dimensions.js:53]: What @aaemnnosttv said. @hussain-t and I just also had a discussion about this internally. I think the `customDimensions` state here will not add any value, so this can be removed.

Comment (hussain-t) [reply assets/js/modules/analytics-4/datastore/custom-dimensions.js:53]: Thanks @aaemnnosttv and @nfmohit. I have removed the state from the store.

**Comment by hussain-t** [comment]
> Thanks @hussain-t – a few points to address here, but also, let's add a test for this slice as mentioned in the IB. I think is still worth doing even though we aren't creating the `createCustomDimension` action. We'll definitely have other tests to add in future issues, so I think it makes sense to scaffold here, but even a few tests for the fetch action would be great. We don't normally test these but there are a few instances where we do.

Thanks, @aaemnnosttv, I didn't add the tests as we don't add for the fetch stores – given the custom validations in place, testing does seem relevant in this scenario. I've now added a few tests to address this. I appreciate your insights! 👍

File: assets/js/modules/analytics-4/datastore/custom-dimensions.js
Comment (hussain-t) [reply assets/js/modules/analytics-4/datastore/custom-dimensions.js]: Updated ✅

**Review by aaemnnosttv — APPROVED** [review]
Great, thanks @hussain-t !
E2E is failing, but this seems unrelated since the code added here is not being used so this must be from a different change.

**Pull request merged**
